### DISCUSSION TITLE
RBAC

### DISCUSSION THREAD

**Discussion by FerdinaKusumah** [ory/keto#868] [Q&A] [answered]
I have question how to implement RBAC using ory keto.

I'm following this tutorial in [here](https://gruchalski.com/posts/2021-05-15-rbac-with-ory-keto/)
even it's not same exact, i've success follow this tutorial using latest `ory keto`

So the point is, i've flow something like this.
user -> (have many role) -> role -> (have many permisions) -> permissions like this

![RBAC flow](https://user-images.githubusercontent.com/19812460/164249513-eec7c980-86d0-448d-a1c8-adb0e79efb53.png)

For simplicity is like this
![simple case](https://user-images.githubusercontent.com/19812460/164249593-b3b87b87-d2d4-49b7-927e-37411039e28c.png)

The question is how can i fetch permissions for every users instead of fetching roles users ??
Or maybe there's any explanation for this case ?

Thank you very much

**Comment by gen1us2k** [comment, marked as answer]
Hello. You can find RBAC-related documentation in [this pull request](https://github.com/ory/docs/pull/749). It explains how you can implement RBAC using Keto (with some workarounds). However,  native support is still work in progress. Follow the progress in
[this issue](https://github.com/ory/keto/issues/598)
